default.py

```diff
@@ -49,15 +49,35 @@ def addon_log(string):
     msg = '%s: %s' % (LOGGING_PREFIX, string)
     xbmc.log(msg=msg, level=xbmc.LOGDEBUG)
 
+
 def show_busy_dialog():
     busydialog.create()
 
+
 def hide_busy_dialog():
     try:
         busydialog.close()
-    except RuntimeError,e:
+    except RuntimeError, e:
         addon_log('Error closing busy dialog: %s' % e.message)
 
+
+def ensure_login_provided():
+    global username
+    global password
+
+    if username == "" or password == "":
+        dialog = xbmcgui.Dialog()
+        answer = dialog.ok(language(30021), language(30050))
+
+        if answer:
+            xbmcaddon.Addon(addon.getAddonInfo('id')).openSettings()
```

Comment:
Are you sure? Because it throws an error here:

>Error Type: <type 'exceptions.TypeError'>
Error Contents: descriptor 'openSettings' of 'xbmcaddon.Addon' object needs an argument
Traceback (most recent call last):
File "..\Kodi\addons\plugin.video.nfl.gamepass\default.py", line 587, 
in <module> xbmcaddon.Addon.openSettings()
TypeError: descriptor 'openSettings' of 'xbmcaddon.Addon' object needs an argument

I tried your exact suggestion:
`xbmcaddon.Addon.openSettings()`